Kernel/Arch/aarch64/pre_init.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023, Timon Kruiper <timonkruiper@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <Kernel/Arch/aarch64/CPU.h>
+#include <Kernel/Sections.h>
+
+// We arrive here from boot.S with the MMU disabled and in an unknown exception level (EL).
+// The kernel is linked at the virtual address, so we have to be really carefull when accessing
+// global variables, as the MMU is not yet enabled.
+
+// FIXME: This should probably be shared with the Prekernel.
+
+namespace Kernel {
+
+extern "C" [[noreturn]] void init();
+
+extern "C" [[noreturn]] void pre_init();
+extern "C" [[noreturn]] void pre_init()
+{
+    // We want to drop to EL1 as soon as possible, because that is the
+    // exception level the kernel should run at.
+    drop_to_exception_level_1();
+
+    // Next step is to setup page tables and enable the MMU.
+    init_page_tables();
+
+    // At this point the MMU is enabled, physical memory is identity mapped,
+    // and the kernel is also mapped into higher virtual memory, however we are still executing
```

Comment:
nit: consider turning this sentence into several. (e.g. ", however …" -> ". However, …". And maybe there's even enough words for 3 sentences)